lib/ansible/modules/database/postgresql/postgresql_privs.py

```diff
@@ -295,6 +295,20 @@ class Error(Exception):
     pass
 
 
+def role_exists(module, cursor, rolname):
+    """Check user exists or not"""
+    query = "SELECT 1 FROM pg_roles WHERE rolname = '%s'" % rolname
```

Comment:
I think you should use the [pg_quote_identifier](https://github.com/ansible/ansible/blob/devel/lib/ansible/module_utils/database.py#L114) here to ensure the query is safe.

```suggestion
    query = "SELECT 1 FROM pg_roles WHERE rolname = '%s'" %  pg_quote_identifier(rolname, "role")
```

Reply:
I can do it.
However, IMO, we don't need to check all possible user input mistakes (like that a name of a database object can't contain more than 63 symbols, etc.). So, I suggest to abandon to use this function (and, respectively, any other functions that check user input for typos, mistakes, etc.).
In any case, seems nothing scary will happen and the PostgreSQL server just will return an exception to psycopg2 and user see it.
I've never broken the Postgres instance by doing mistakes in object names ;)